test/hotspot/jtreg/runtime/ErrorHandling/TestSigInfoInHsErrFile.java

```diff
@@ -68,7 +68,9 @@ public static void main(String[] args) throws Exception {
     patterns.add(Pattern.compile("# .*VMError::controlled_crash.*"));
 
     // Crash address: see VMError::_segfault_address
-    String crashAddress = Platform.isAix() ? "0xffffffffffffffff" : "0x0*400";
+    //All faults on s390x give the address only on page granularity.
+    //Hence fault address is first page address.
+    String crashAddress = Platform.isAix() ? "0xffffffffffffffff" : Platform.isS390x() ? "0x0*1000" : "0x0*400";
```

Comment:
Probably we can create a if-else ladder here and move this comment inside that if/else block ? something like this: 
```
if (AIX) {
crashAddress for AIX
} else if (S390) {
// your comment
crashAddress for S390
} else {
crashAddress for other archs
}
```

Reply:
I have adapted this.